libtransmission/torrent.cc

```diff
@@ -2568,8 +2568,8 @@ void renameTorrentFileString(tr_torrent* tor, std::string_view oldpath, std::str
 
 void torrentRenamePath(
     tr_torrent* const tor,
-    std::string oldpath, // NOLINT performance-unnecessary-value-param
-    std::string newname, // NOLINT performance-unnecessary-value-param
+    std::string const& oldpath, // NOLINT performance-unnecessary-value-param
```

Comment:
Actually we might just change `old_path` and `new_path` to both be `std::string_view` here in `torrentRenamePath()` I think?

Reply:
You could try changing the signature of `tr_torrent_rename_done_func` to take a `std::string_view` and then remove the `.c_str()`.

Reply:
:dizzy: @Coeur weren't you wanting to keep c++isms out of transmission.h to make Swift migration easier?

> You could try changing the signature of `tr_torrent_rename_done_func` to take a `std::string_view` and then remove the `.c_str()`.

Let's not do that; that escalates a perf refactor into a breaking change since transmission.h is public API.

@midzer is right, my original suggestion doesn't work with the original function signature :man_shrugging:

Reply:
Okay, let me revert last commit.

Reply:
yes, removing C++ is one of my goals, but it won't happen anytime soon: it might be fine to live with two headers: one for C++, and one which is a C wrapper with compatibility shims (similar to the one I made in my branch).